<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsMeshCalculatorDialogBase</class>
 <widget class="QDialog" name="QgsMeshCalculatorDialogBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>912</width>
    <height>982</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Mesh Calculator</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QgsScrollArea" name="scrollArea_2">
     <property name="frameShape">
      <enum>QFrame::Shape::NoFrame</enum>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>900</width>
        <height>906</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QSplitter" name="splitter_2">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <widget class="QSplitter" name="splitter">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <widget class="QGroupBox" name="mRasterBandsGroupBox">
           <property name="title">
            <string>Datasets</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="0">
             <widget class="QListView" name="mDatasetsListWidget"/>
            </item>
           </layout>
          </widget>
          <widget class="QGroupBox" name="mResultGroupBox">
           <property name="title">
            <string>Result Layer</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QgsScrollArea" name="scrollArea">
              <property name="frameShape">
               <enum>QFrame::Shape::NoFrame</enum>
              </property>
              <property name="widgetResizable">
               <bool>true</bool>
              </property>
              <widget class="QWidget" name="scrollAreaWidgetContents">
               <property name="geometry">
                <rect>
                 <x>0</x>
                 <y>0</y>
                 <width>509</width>
                 <height>406</height>
                </rect>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_6">
                <property name="leftMargin">
                 <number>0</number>
                </property>
                <property name="topMargin">
                 <number>0</number>
                </property>
                <property name="rightMargin">
                 <number>0</number>
                </property>
                <property name="bottomMargin">
                 <number>0</number>
                </property>
                <item>
                 <layout class="QGridLayout" name="gridLayout_5">
                  <item row="3" column="0">
                   <widget class="QLabel" name="mOutputFormatLabel_2">
                    <property name="text">
                     <string>Group name</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="0" colspan="3">
                   <widget class="QCheckBox" name="mUseVirtualProviderCheckBox">
                    <property name="layoutDirection">
                     <enum>Qt::LayoutDirection::LeftToRight</enum>
                    </property>
                    <property name="text">
                     <string>Create on-the-fly dataset group instead of writing layer to disk</string>
                    </property>
                    <property name="tristate">
                     <bool>false</bool>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1" colspan="2">
                   <widget class="QgsFileWidget" name="mOutputDatasetFileWidget" native="true"/>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLabel" name="mOutputDatasetFileLabel">
                    <property name="text">
                     <string>Output file</string>
                    </property>
                   </widget>
                  </item>
                  <item row="3" column="1" colspan="2">
                   <widget class="QLineEdit" name="mOutputGroupNameLineEdit"/>
                  </item>
                  <item row="2" column="0">
                   <widget class="QLabel" name="mOutputFormatLabel">
                    <property name="text">
                     <string>Output format</string>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="1" colspan="2">
                   <widget class="QComboBox" name="mOutputFormatComboBox"/>
                  </item>
                 </layout>
                </item>
                <item>
                 <widget class="QGroupBox" name="groupBox_3">
                  <property name="title">
                   <string>Spatial Extent</string>
                  </property>
                  <layout class="QVBoxLayout" name="verticalLayout_5">
                   <item>
                    <widget class="QWidget" name="horizontalWidget" native="true">
                     <layout class="QHBoxLayout" name="horizontalLayout_5">
                      <property name="leftMargin">
                       <number>0</number>
                      </property>
                      <property name="topMargin">
                       <number>0</number>
                      </property>
                      <property name="rightMargin">
                       <number>0</number>
                      </property>
                      <property name="bottomMargin">
                       <number>0</number>
                      </property>
                      <item>
                       <widget class="QRadioButton" name="useExtentCb">
                        <property name="text">
                         <string>Custom extent</string>
                        </property>
                        <property name="checked">
                         <bool>true</bool>
                        </property>
                        <attribute name="buttonGroup">
                         <string notr="true">buttonGroup</string>
                        </attribute>
                       </widget>
                      </item>
                      <item>
                       <widget class="QRadioButton" name="useMaskCb">
                        <property name="toolTip">
                         <string>Clips the datasets using features from vector polygon layer.</string>
                        </property>
                        <property name="text">
                         <string>Mask layer</string>
                        </property>
                        <attribute name="buttonGroup">
                         <string notr="true">buttonGroup</string>
                        </attribute>
                       </widget>
                      </item>
                     </layout>
                    </widget>
                   </item>
                   <item>
                    <widget class="QWidget" name="maskBox" native="true">
                     <layout class="QHBoxLayout" name="maskBoxLayout" stretch="0,0">
                      <property name="topMargin">
                       <number>0</number>
                      </property>
                      <property name="bottomMargin">
                       <number>0</number>
                      </property>
                      <item>
                       <widget class="QLabel" name="label_3">
                        <property name="minimumSize">
                         <size>
                          <width>206</width>
                          <height>0</height>
                         </size>
                        </property>
                        <property name="text">
                         <string>Mask layer</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QgsMapLayerComboBox" name="cboLayerMask">
                        <property name="sizePolicy">
                         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                          <horstretch>0</horstretch>
                          <verstretch>0</verstretch>
                         </sizepolicy>
                        </property>
                        <property name="editable">
                         <bool>false</bool>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </widget>
                   </item>
                   <item>
                    <widget class="QgsExtentGroupBox" name="mExtentGroupBox">
                     <property name="title">
                      <string>Spatial Extent</string>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item>
                 <widget class="QGroupBox" name="groupBox_2">
                  <property name="title">
                   <string>Temporal Extent</string>
                  </property>
                  <layout class="QGridLayout" name="gridLayout_6">
                   <item row="1" column="2">
                    <spacer name="horizontalSpacer">
                     <property name="orientation">
                      <enum>Qt::Orientation::Horizontal</enum>
                     </property>
                     <property name="sizeHint" stdset="0">
                      <size>
                       <width>10</width>
                       <height>20</height>
                      </size>
                     </property>
                    </spacer>
                   </item>
                   <item row="1" column="3">
                    <widget class="QLabel" name="label_2">
                     <property name="text">
                      <string>End time</string>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="0">
                    <widget class="QLabel" name="label">
                     <property name="text">
                      <string>Start time</string>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="1">
                    <widget class="QComboBox" name="mStartTimeComboBox"/>
                   </item>
                   <item row="1" column="4">
                    <widget class="QComboBox" name="mEndTimeComboBox"/>
                   </item>
                   <item row="0" column="0" colspan="5">
                    <layout class="QHBoxLayout" name="horizontalLayout_2">
                     <item>
                      <widget class="QPushButton" name="mAllTimesButton">
                       <property name="text">
                        <string>Use all Selected Dataset Times</string>
                       </property>
                      </widget>
                     </item>
                     <item>
                      <spacer name="horizontalSpacer_2">
                       <property name="orientation">
                        <enum>Qt::Orientation::Horizontal</enum>
                       </property>
                       <property name="sizeHint" stdset="0">
                        <size>
                         <width>40</width>
                         <height>20</height>
                        </size>
                       </property>
                      </spacer>
                     </item>
                    </layout>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item>
                 <widget class="QCheckBox" name="mAddResultToProjectCheckBox">
                  <property name="text">
                   <string>Add result to project</string>
                  </property>
                  <property name="checked">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
         <widget class="QWidget" name="verticalLayoutWidget">
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QgsCollapsibleGroupBox" name="mOperatorsGroupBox">
             <property name="title">
              <string>Operators</string>
             </property>
             <layout class="QGridLayout" name="gridLayout">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item row="0" column="0">
               <widget class="QPushButton" name="mPlusPushButton">
                <property name="text">
                 <string>+</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QPushButton" name="mMultiplyPushButton">
                <property name="text">
                 <string>*</string>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QPushButton" name="mOpenBracketPushButton">
                <property name="text">
                 <string>(</string>
                </property>
               </widget>
              </item>
              <item row="0" column="3">
               <widget class="QPushButton" name="mMinButton">
                <property name="text">
                 <string>min</string>
                </property>
               </widget>
              </item>
              <item row="0" column="4">
               <widget class="QPushButton" name="mIfButton">
                <property name="text">
                 <string>IF</string>
                </property>
               </widget>
              </item>
              <item row="0" column="5">
               <widget class="QPushButton" name="mSumAggrButton">
                <property name="text">
                 <string>sum (aggr)</string>
                </property>
               </widget>
              </item>
              <item row="0" column="6">
               <spacer name="horizontalSpacer_1">
                <property name="orientation">
                 <enum>Qt::Orientation::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="1" column="0">
               <widget class="QPushButton" name="mMinusPushButton">
                <property name="text">
                 <string>-</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QPushButton" name="mDividePushButton">
                <property name="text">
                 <string>/</string>
                </property>
               </widget>
              </item>
              <item row="1" column="2">
               <widget class="QPushButton" name="mCloseBracketPushButton">
                <property name="text">
                 <string>)</string>
                </property>
               </widget>
              </item>
              <item row="1" column="3">
               <widget class="QPushButton" name="mMaxButton">
                <property name="text">
                 <string>max</string>
                </property>
               </widget>
              </item>
              <item row="1" column="4">
               <widget class="QPushButton" name="mAndButton">
                <property name="text">
                 <string>AND</string>
                </property>
               </widget>
              </item>
              <item row="1" column="5">
               <widget class="QPushButton" name="mMaxAggrButton">
                <property name="text">
                 <string>max (aggr)</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QPushButton" name="mLessButton">
                <property name="text">
                 <string>&lt;</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QPushButton" name="mGreaterButton">
                <property name="text">
                 <string>&gt;</string>
                </property>
               </widget>
              </item>
              <item row="2" column="2">
               <widget class="QPushButton" name="mEqualButton">
                <property name="text">
                 <string>=</string>
                </property>
               </widget>
              </item>
              <item row="2" column="3">
               <widget class="QPushButton" name="mAbsButton">
                <property name="text">
                 <string>abs</string>
                </property>
               </widget>
              </item>
              <item row="2" column="4">
               <widget class="QPushButton" name="mOrButton">
                <property name="text">
                 <string>OR</string>
                </property>
               </widget>
              </item>
              <item row="2" column="5">
               <widget class="QPushButton" name="mMinAggrButton">
                <property name="text">
                 <string>min (aggr)</string>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QPushButton" name="mLesserEqualButton">
                <property name="text">
                 <string>&lt;=</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QPushButton" name="mGreaterEqualButton">
                <property name="text">
                 <string>&gt;=</string>
                </property>
               </widget>
              </item>
              <item row="3" column="2">
               <widget class="QPushButton" name="mNotEqualButton">
                <property name="text">
                 <string>!=</string>
                </property>
               </widget>
              </item>
              <item row="3" column="3">
               <widget class="QPushButton" name="mPowButton">
                <property name="text">
                 <string>^</string>
                </property>
               </widget>
              </item>
              <item row="3" column="4">
               <widget class="QPushButton" name="mNotButton">
                <property name="text">
                 <string>NOT</string>
                </property>
               </widget>
              </item>
              <item row="3" column="5">
               <widget class="QPushButton" name="mAverageAggrButton">
                <property name="text">
                 <string>average (aggr)</string>
                </property>
               </widget>
              </item>
              <item row="4" column="0" colspan="6">
               <widget class="QPushButton" name="mNoDataButton">
                <property name="text">
                 <string>NODATA</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox">
             <property name="title">
              <string>Mesh Calculator Expression</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QTextEdit" name="mExpressionTextEdit"/>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="mExpressionValidLabel">
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="mButtonBox">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::StandardButton::Cancel|QDialogButtonBox::StandardButton::Help|QDialogButtonBox::StandardButton::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
  <zorder>mButtonBox</zorder>
  <zorder>mExpressionValidLabel</zorder>
  <zorder>scrollArea_2</zorder>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>qgscollapsiblegroupbox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsFileWidget</class>
   <extends>QWidget</extends>
   <header>qgsfilewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsExtentGroupBox</class>
   <extends>QGroupBox</extends>
   <header>qgsextentgroupbox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsScrollArea</class>
   <extends>QScrollArea</extends>
   <header>qgsscrollarea.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsMapLayerComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsmaplayercombobox.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>mDatasetsListWidget</tabstop>
  <tabstop>mUseVirtualProviderCheckBox</tabstop>
  <tabstop>mOutputFormatComboBox</tabstop>
  <tabstop>mOutputGroupNameLineEdit</tabstop>
  <tabstop>useExtentCb</tabstop>
  <tabstop>useMaskCb</tabstop>
  <tabstop>cboLayerMask</tabstop>
  <tabstop>mAllTimesButton</tabstop>
  <tabstop>mStartTimeComboBox</tabstop>
  <tabstop>mEndTimeComboBox</tabstop>
  <tabstop>mPlusPushButton</tabstop>
  <tabstop>mMultiplyPushButton</tabstop>
  <tabstop>mOpenBracketPushButton</tabstop>
  <tabstop>mMinButton</tabstop>
  <tabstop>mIfButton</tabstop>
  <tabstop>mSumAggrButton</tabstop>
  <tabstop>mMinusPushButton</tabstop>
  <tabstop>mDividePushButton</tabstop>
  <tabstop>mCloseBracketPushButton</tabstop>
  <tabstop>mMaxButton</tabstop>
  <tabstop>mAndButton</tabstop>
  <tabstop>mMaxAggrButton</tabstop>
  <tabstop>mLessButton</tabstop>
  <tabstop>mGreaterButton</tabstop>
  <tabstop>mEqualButton</tabstop>
  <tabstop>mAbsButton</tabstop>
  <tabstop>mOrButton</tabstop>
  <tabstop>mMinAggrButton</tabstop>
  <tabstop>mLesserEqualButton</tabstop>
  <tabstop>mGreaterEqualButton</tabstop>
  <tabstop>mNotEqualButton</tabstop>
  <tabstop>mPowButton</tabstop>
  <tabstop>mNotButton</tabstop>
  <tabstop>mAverageAggrButton</tabstop>
  <tabstop>mNoDataButton</tabstop>
  <tabstop>mExpressionTextEdit</tabstop>
  <tabstop>mButtonBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>mButtonBox</sender>
   <signal>accepted()</signal>
   <receiver>QgsMeshCalculatorDialogBase</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mButtonBox</sender>
   <signal>rejected()</signal>
   <receiver>QgsMeshCalculatorDialogBase</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
